.envrc

```diff
@@ -1,26 +1,17 @@
 #!/usr/bin/bash
 set -e
 
-if [ "$(python -V)" != "Python $(cat .python-version)" ]; then
-    echo "WARNING: Your version does not match .python-version. You're on your own."
-    echo "You could use pyenv to match the version."
-fi
+# XXX: PDM requires Python 3.7+; check for python3 version
 
-if ! command -v pdm &>/dev/null; then
-    curl -sSL https://raw.githubusercontent.com/pdm-project/pdm/main/install-pdm.py | python -
+if [ ! -f "${HOME}/.local/bin/pdm" ]; then
+    curl -sSL https://raw.githubusercontent.com/pdm-project/pdm/main/install-pdm.py | python3 -
     echo "export PATH=${HOME}/.local/bin:\$PATH" >> ~/.zshrc
     export PATH="${HOME}/.local/bin:$PATH"
-    pdm --pep582 >>~/.zshrc
     pdm plugin add pdm-venv
 fi
 
-eval "$(pdm venv activate 3.8)"
-# If it fails to activate is because we need to create it
-if [ -z "$VIRTUAL_ENV" ]; then
-    (pdm venv list | grep 3.8) || pdm venv create 3.8
-    eval "$(pdm venv activate 3.8)"
-fi
-
-echo "Run pdm sync if you want the dependencies to get installed"
+! command -v pdm &>/dev/null && export PATH="${HOME}/.local/bin:$PATH"
```

Comment:
There is a chance that VS Code is initialized before pdm is placed in the path. This handles that case.